python/lsst/ctrl/mpexec/cmdLineParser.py

```diff
@@ -381,6 +382,10 @@ def makeParser(fromfile_prefix_chars='@', parser_class=ArgumentParser, **kwargs)
         subparser.add_argument("-C", "--configfile", metavar="LABEL:PATH",
                                dest="pipeline_actions", action='append', type=_ACTION_CONFIG_FILE,
                                help="Configuration override file(s), applies to a task with a given label.")
+        subparser.add_argument("--dataset-name-substitution", metavar="LABEL:VALUE",
+                               dest='pipeline_actions', action='append', type=_ACTION_NAME_TEMPLATES,
+                               help='Configuration name template values, applies to a task with a given '
+                                    'label.')
```

Comment:
Looking at the code I think VALUE in this case is a Python mapping of string to string. It may be rather messy to write that on command line and will need a lot of quoting, I suppose it should look like:

    --dataset-name-substitution 'FancyTask:{"inputCoaddName": "deep2", "outputCoaddName": "deep2"}'

Could some other format with less quoting be more user-friendly, maybe something like:

    --dataset-name-substitution FancyTask:inputCoaddName=deep2,outputCoaddName=deep2

Reply:
Jim and I debated this a little bit with pros and cons for several different ways. The conclusion we reached is that for now we will leave it as is, and collect data from several people in the group for the input pattern they would prefer, and make any changes in a future ticket.